7a_temp_coop_munge/src/data_parsers/parse_Iowa_DNR_LimnoProfiles_2000_2020.R

```diff
@@ -0,0 +1,507 @@
+    if(check_final_depth(dat$depth)) {
+      depth_loc <- which(names(dat) == 'depth') # id position of the depth column
+      dat <- trim_to_max_depth(dat, depth_col = depth_loc)
+
+      # return(dat)
+    }
+  } else {
+    print(paste('File ', basename(x),
+                ' appears to be empty or has problematic data. Data only has ',
+                dat_rows, ' row(s) of data', sep = ''))
+  }
+
+  return(dat)
+}
+
+#' Determine if a file is `xls` or `htm`
+#'
+#' Data parser for data from 2010-2016. These files are in a variety of
+#' mismatched formats. this parser can differentiate between `xls` and
+#' `htm` (labeled as `xls`)
+#'
+#' @param x path, full file path
+#' @param keep_cols chr, vector of columns names that should be preserved
+#' @param subset_lakeid num, vector for start/stop values passed to `base::substr` in `extract_lakeid`
+#' Column name spelling should match. Case does not matter because
+#' all values to be changed `tolower` within the function.
+#' @param htm_datetime logical, `TRUE` if the file is expected to have
+#' a `datetime` field. This is not always the case in the 2016 data. Argument
+#' used if data is `htm` format.
+#' @param htm_data_start num, on what row does the data start? Argument used if
+#' data is `htm` format.
+#' @param htm_verbose logical, if `TRUE` a message that identifies an HTM file
+#' is returned. Argument used if data is `htm` format.
+#'
+parse_2010_2016_data <- function(x, keep_cols, subset_lakeid, htm_datetime = F,
+                                 htm_data_start = 2, htm_verbose = F) {
+  tc_out <- tryCatch(
+    { read_iowa_xls_tbl(x, keep_cols, data_start = 3) }, # current thinking is that all true xls start on row 3
+    error = function(cond) {
+      if(htm_verbose) {print(paste(basename(x), ' is in HTM format', sep = ''))}
+      read_iowa_htm_tbl(x, keep_cols, htm_datetime, htm_data_start)
+    }
+  )
+
+  # add LakeID and date columns from file name if necessary
+  tc_out$lakeid <- extract_lakeid(x, start = subset_lakeid[1], stop = subset_lakeid[2])
+
+  return(tc_out)
+}
+
+#' Parse Iowas data from 2017-2020
+#'
+#' @param x chr, full file path
+#' @param keep_cols chr, vector of column names to keep.
+#' @param lakeid_col num, Column index for the location of the Lake ID field. The column that is used to ID the Iowa lake moves around and changes names from file to file.
+#' @param temp_as_f logical, should temp values be converted from F to C?
+#' @param use_readr logical, not all files are read well by `readr::read_csv`. If `use_readr = F` then `read.csv` is used.
+#' Column name spelling should match. Case does not matter because
+#' all values to be changed `tolower` within the function.
+parse_2017_2020_data <- function(x, keep_cols, lakeid_col, temp_as_f = F, use_readr = T){
+
+  if(grepl('*xlsx', x)) {
+    dat <- readxl::read_xlsx(x)
+  } else if (use_readr){
+    dat <- readr::read_csv(x)
+  } else {
+    dat <- read.csv(x)
```

Comment:
Would you be up for changing this `read.csv` to `read.csv(x, fileEncoding="latin1")`? Or does that fail on your windows machine for "Iowa_DNR_LimnoProfiles_2000-2020/Iowa_profiles_2017b/17129263ysi.csv"?

Reply:
See https://github.com/padilla410/lake-temperature-model-prep/pull/1

Reply:
Got it. I added it to the function and tested it on my end. I am going to reject your PR into my repo because I will resolve it here.